### DISCUSSION TITLE
Polling faster than every 10 seconds

### DISCUSSION THREAD

**Discussion by nis** [wlcrs/huawei_solar#434] [Setting up the integration] [answered]
I'm trying to get polling going faster than every 10s as I have a EV charger that needs the current data from the smartmeter connected to the inverter. The faster the updates, the smoother the charging.

I've set up an automation with manual polling to run every 2 seconds, but the data only gets updated every 12-15 seconds.

Is there some sort of limit in Home Assistant, The inverter, Smart Meter, this integration or somewhere else I should be aware of?

I can control the inverter and sees the same data as the Fusion App, so the communication is working. I would just like it to update faster :)

**Comment by Ptrancoso84** [comment]
Hi

i have mine like this without any issues

create an area with the inverter/battery/meter an then an automation:

in this case, the area is called "solar_energy"

alias: Update Solar Energy
description: ""
trigger:
  - platform: time_pattern
    hours: "*"
    minutes: "*"
    seconds: "1"
condition: []
action:
  - service: homeassistant.update_entity
    data: {}
    target:
      area_id: solar_energy
mode: single


PS: HA takes some time to process all data, and the values only update if they change, for exemple if solar production is 500w for 20s it will say "updated 20s ago", despite its been updating the value every 2s

**Reply by nis** [reply]
I got the automation doing the polling set up. At least the GUI says: "Trigger every 2 seconds of every minute of every hour". Also is only says "now" in the "last triggered" column.

But the data is only updated every 12 seconds: ![image](https://github.com/wlcrs/huawei_solar/assets/2558/6f302b31-b42a-47f6-8d82-88e7ead9426f)

**Reply by ruteclrp** [reply, marked as answer]
I think the MODBUS protocol runs as a sequential reading of one register at the time and with the large number of entities to be updated, this just takes time. 
I am also unable to get data updated more frequently than every 12 sec. I have set the polling frequency to 20 sec to avoid overloading the connection.
